### PR TITLE
PE-4319: implementation of entity generator component for large files

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/01pvpj5v5vno0

### CODE REVIEW

File: lib/core/upload/metadata_generator.dart [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import 'package:ardrive/core/arfs/entities/arfs_entities.dart' as arfs;
+import 'package:ardrive/core/upload/upload_metadata.dart';
+import 'package:ardrive/entities/constants.dart';
+import 'package:ardrive/services/app/app_info_services.dart';
+import 'package:ardrive_io/ardrive_io.dart';
+import 'package:arweave/arweave.dart';
+import 'package:uuid/uuid.dart';
+
+/// this class will get an `IOFile` and mounts the metadata for it
```
Comment (karlprieb) [inline lib/core/upload/metadata_generator.dart:9]:
```suggestion
/// this class will get an `IOFile` and generate the metadata for it
```

File: lib/core/upload/metadata_generator.dart [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    final id = const Uuid().v4();
+
+    if (entity is IOFile) {
+      ARFSUploadMetadataArgsValidator.validate(arguments, arfs.EntityType.file);
+
+      final file = entity;
+
+      return ARFSFileUploadMetadata(
```
Comment (karlprieb) [inline lib/core/upload/metadata_generator.dart:55]: you're missing privacy

File: lib/core/upload/metadata_generator.dart [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    return tags;
+  }
+
+  List<Tag> get _appTags {
+    final appInfo = _appInfoServices.appInfo;
+
+    final appVersion = Tag(EntityTag.appVersion, appInfo.version);
+    final appPlatform = Tag(EntityTag.appPlatform, appInfo.platform);
+    final arfsTag = Tag(EntityTag.arFs, appInfo.arfsVersion);
+    final entityTag = Tag(EntityTag.entityType, _entity.name);
```
Comment (karlprieb) [inline lib/core/upload/metadata_generator.dart:192]: remove this one as it's being handled by `_entityTags`

File: lib/core/upload/metadata_generator.dart [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+      case arfs.EntityType.folder:
+        if (args.driveId == null) {
+          throw ArgumentError('driveId must not be null');
+        }
+        break;
+
+      case arfs.EntityType.drive:
+        if (args.privacy == null) {
+          throw ArgumentError('privacy must not be null');
+        }
```
Comment (karlprieb) [inline lib/core/upload/metadata_generator.dart:238]: move privacy and driveId check for outside of the switch as you need to check that for all entity types

Comment (thiagocarvalhodev) [reply lib/core/upload/metadata_generator.dart:243]: We don't add the privacy tag for other entities but drive.

File: lib/core/upload/metadata_generator.dart (hunk 2 above)
Comment (thiagocarvalhodev) [reply lib/core/upload/metadata_generator.dart:55]: Added the property: `isPrivate` to `ARFSFileUploadMetadata`

**Review by karlprieb — APPROVED** [review]

**Pull request merged**